### PR TITLE
Removed StoreFactory since it does not provide much utility.

### PR DESCRIPTION
**Pull request by bdmogal**
Build: https://builds.cask.co/browse/CDAP-DUT1272

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java [hunk 1]
```diff
@@ -266,10 +265,7 @@ protected void configure() {
           .build(new TypeLiteral<ManagerFactory<DeploymentInfo, ApplicationWithPrograms>>() { })
       );
 
-      install(new FactoryModuleBuilder()
-                .implement(Store.class, DefaultStore.class)
-                .build(StoreFactory.class)
-      );
+      bind(Store.class).to(DefaultStore.class).in(Scopes.SINGLETON);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: This change the behavior. When factory was used, a new instance is created every time upon request. Do you know if `Store` is safe to be a singleton?

Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: Store is not thread-safe... I think Terence is right and we need to provide a new instance every time.

**Comment by anew** [comment]
@bdmogal I understand what you are trying to do. I do not think this is sufficient. Creating a new instance every time it is injected (instead of using a singleton) is not the same as creating a new instance every time it is used. If you inject this into a multi-threaded class, such as an HTTP handler, each thread will still use the same instance that was injected. If you inject a provider that creates a new instance every time it is invoked, makes every thread use a different instance.

Can we just drop this and leave it the way it was? :)

**Comment by chtyim** [comment]
I think for all current usages of storefactory, they only do the create call in constructor, hence making the existence of factory class unnecessary. Also, in guice, for factory.create method that doesn't take any parameter, we better use the guice Provider instead if there is a need to create new instance besides in @Inject constructor. Provider binding are available by default for all bindings

**Comment by anew** [comment]
I looked at the code a little more, and I concur with Terence. Good to me if you use the provider.

**Comment by chtyim** [comment]
You don't need to use provider when the injection happens in construction.

**Comment by chtyim** [comment]
LGTM

**Comment by bdmogal** [comment]
Build passed - http://builds.cask.co/browse/CDAP-DUT1272-7

Merging.

**Pull request merged**